File: CreationalPatterns/AbstractFactory/C#/Vehicle Factory/ConcreteFactories/GermanVehicleFactory.cs
using Vehicles.Interfaces;
using Vehicles.Products;

namespace Vehicles.ConcreteFactories
{
    public class GermanVehicleFactory : IVehicleFactory
    {
        public IVehicle CreateSedan()
        {
            return new GermanSedan();
        }

        public IVehicle CreateSUV()
        {
            return new GermanSUV();
        }

        public IVehicle CreateSportsCar()
        {
            return new GermanSportsCar();
        }
    }
}

File: CreationalPatterns/AbstractFactory/Java/Vehicle/src/concretefactories/GermanVehicleFactory.java
package concretefactories;

import interfaces.IVehicle;
import interfaces.IVehicleFactory;
import products.GermanSUV;
import products.GermanSedan;
import products.GermanSportsCar;

public class GermanVehicleFactory implements IVehicleFactory {

    @Override
    public IVehicle createSedan() {
        return new GermanSedan();
    }

    @Override
    public IVehicle createSUV() {
        return new GermanSUV();
    }

    @Override
    public IVehicle createSportsCar() {
        return new GermanSportsCar();
    }
}